webviz_subsurface/private_containers/_surface_selector.py

```diff
@@ -0,0 +1,564 @@
+            options = [{"value": sens, "label": sens} for sens in sens_names]
+            return options, value, self.show_dropdown_style
+
+        @app.callback(
+            [
+                Output(self.sens_case_id, "options"),
+                Output(self.sens_case_id, "value"),
+                Output(self.sens_case_wrapper_id, "style"),
+            ],
+            [Input(self.sens_name_id, "value")],
+            [State(self.ensemble_id, "value"), State(self.sens_case_id, "value")],
+        )
+        def _update_sens_case(sensname, ensemble, current_value):
+            sens_cases = self.sens_cases(ensemble, sensname)
+            if not sens_cases:
+                return [], None, self.hide_dropdown_style
+            value = current_value if current_value in sens_cases else sens_cases[0]
+            options = [{"value": sens, "label": sens} for sens in sens_cases]
+            return options, value, self.show_dropdown_style
+
+        @app.callback(
+            Output(self.storage_id, "children"),
+            [
+                Input(self.attr_id, "value"),
+                Input(self.name_id, "value"),
+                Input(self.date_id, "value"),
+                Input(self.ensemble_id, "value"),
+                Input(self.aggreal_id, "value"),
+                Input(self.realization_id, "value"),
+                Input(self.sens_name_id, "value"),
+                Input(self.sens_case_id, "value"),
+            ],
+        )
+        def _set_data(
+            attr, name, date, ensemble, aggreal, calculation, sens_name, sens_case
+        ):
+            reals = self.realizations(ensemble, sens_name, sens_case)
+            return json.dumps(
+                {
+                    "attribute": attr,
+                    "name": name,
+                    "date": date,
+                    "ensemble": ensemble,
+                    "aggregation": calculation if aggreal == "Aggregation" else None,
+                    "realization": reals if aggreal == "Aggregation" else calculation,
+                }
+            )
+
+
+def prev_value(current_value, options):
+    try:
+        index = options.index(current_value)
+    except ValueError:
+        index = None
+    if index > 0:
+        return options[index - 1]
+    else:
+        return current_value
+
+
+def next_value(current_value, options):
+    try:
+        index = options.index(current_value)
+    except ValueError:
+        index = None
+    if index < len(options) - 1:
+        return options[index + 1]
+    else:
+        return current_value
+
+
+def format_date(date_string):
+    d = str(date_string)
+    if len(d) == 8:
+        d = datetime.strptime(d, "%Y%m%d").strftime("%m/%d/%Y")
+    if len(d) == 17:
+        begin = d[0:8]
+        end = d[9:17]
+        d = f"({datetime.strptime(begin, '%Y%m%d').strftime('%m/%d/%Y')})-\
+              ({datetime.strptime(end, '%Y%m%d').strftime('%m/%d/%Y')})"
+    return d
```

Comment:
- On purpose to change it from `YYYYMMDD` to `MM/DD/YYYY`?
- If we are doing some formatting here, we could perhaps simplify it such that the container optionally only shows e.g. `February 2020` instead of full date (typically knowing the month is maybe enough wrt. the time resolution of the physical processes in the subsurface). Either as a boolean argument to the container (show day of month or not), or some logic in the container (if delta between dates of same month and year, or if two neighbouring dates are in same month + year --> show day)